### PR TITLE
Feature twotwobugs

### PR DESCRIPTION
**Pull request by planes**
The intention of this PR is to fix bugs but not add features for a 2.2 release.

This PR addresses several issues: #362 #361 #359 #133 and #5 as well as updating a couple of nomenclature problems left over from the PGM PR.

### CODE REVIEW

File: trunk/SUAVE/Methods/Performance/estimate_landing_field_length.py [hunk 1]
```diff
@@ -5,14 +5,16 @@
...
+#           Jul 2020, E. Botero 
 
 # ----------------------------------------------------------------------
 #  Imports
 # ----------------------------------------------------------------------
 
 import SUAVE
-from   SUAVE.Core            import Data
-from   SUAVE.Core            import Units
+from   SUAVE.Core import Data
```
Comment (timdmacdo) [inline trunk/SUAVE/Methods/Performance/estimate_landing_field_length.py:15]: Formatting looks odd here

File: trunk/SUAVE/Methods/Performance/estimate_take_off_field_length.py [hunk 2]
```diff
@@ -187,9 +182,11 @@ def estimate_take_off_field_length(vehicle,analyses,airport,compute_2nd_seg_clim
...
     if compute_2nd_seg_climb:
         # Getting engine thrust at V2 (update only speed related conditions)
-        state.conditions.freestream.dynamic_pressure = np.array(np.atleast_1d(0.5 * rho * V2_speed**2))
-        state.conditions.freestream.velocity         = np.array(np.atleast_1d(V2_speed))
-        state.conditions.freestream.mach_number      = np.array(np.atleast_1d(V2_speed/ a))
+        state.conditions.freestream.dynamic_pressure  = np.array(np.atleast_1d(0.5 * rho * V2_speed**2))
+        state.conditions.freestream.velocity          = np.array(np.atleast_1d(V2_speed))
+        state.conditions.freestream.mach_number       = np.array(np.atleast_1d(V2_speed/ a))
+        state.conditions.freestream.dynamic_viscosity = np.array(np.atleast_1d(mu))
+        state.conditions.freestream.density          =  np.array(np.atleast_1d(rho))
```
Comment (timdmacdo) [inline trunk/SUAVE/Methods/Performance/estimate_take_off_field_length.py:189]: Missed an equal sign indentation here.

**Comment by planes** [comment]
> Looks pretty good. Do the changes to the ground segments return them to the older form?

In some ways yes, I pulled in some of your changes from the PGM_weights_v2 branch. However, it didn't work out of the box. So I tweaked it a bit. It seems be working appropriately now.

File: trunk/SUAVE/Methods/Performance/estimate_landing_field_length.py (hunk 1 above)
Comment (planes) [reply trunk/SUAVE/Methods/Performance/estimate_landing_field_length.py]: I think I fixed this?

File: regression/scripts/Regional_Jet_Optimization/Procedure2.py [hunk 3]
```diff
@@ -164,43 +165,47 @@ def simple_sizing(nexus):
...
     #landing.store_diff()
     
     
     #Takeoff CL_max
     takeoff = nexus.vehicle_configurations.takeoff
-    takeoff_conditions = Data()
-    takeoff_conditions.freestream = Data()    
+    state = Data()
+    state.conditions = Data()
+    state.conditions.freestream = Data()
```
Comment (mclarke2) [inline regression/scripts/Regional_Jet_Optimization/Procedure2.py:183]: Table format

File: trunk/SUAVE/Input_Output/OpenVSP/vsp_write.py [hunk 4]
```diff
@@ -677,6 +678,15 @@ def write_vsp_fuselage(fuselage,area_tags, main_wing, fuel_tank_set_ind):
         vsp.SetParmVal(fuse_id, "Ellipse_Height", "XSecCurve_2", height2);
         vsp.SetParmVal(fuse_id, "Ellipse_Height", "XSecCurve_3", height3);  
     else:
+        # OpenVSP vals do not exist:
+        vals = Data()
```
Comment (mclarke2) [inline trunk/SUAVE/Input_Output/OpenVSP/vsp_write.py:682]: table format

File: trunk/SUAVE/Methods/Missions/Segments/Ground/Common.py [hunk 5]
```diff
@@ -87,25 +91,48 @@ def initialize_conditions(segment):
...
+    
+    # Setup the size of the residuals
+    ones_row_m1 = segment.state.ones_row_m1
+    segment.state.residuals.forces = ones_row_m1(1) * 0.0      
 
     # pack conditions 
-    conditions.frames.inertial.velocity_vector[:,0] = np.linspace(v0,vf,N)
-    conditions.ground.incline[:,0]                  = segment.ground_incline
-    conditions.ground.friction_coefficient[:,0]     = segment.friction_coefficient
+    segment.state.conditions.frames.inertial.velocity_vector[:,0] = initialized_velocity[:,0]
```
Comment (mclarke2) [inline trunk/SUAVE/Methods/Missions/Segments/Ground/Common.py:130]: table format

File: trunk/SUAVE/Methods/Missions/Segments/Ground/Common.py [hunk 6]
```diff
@@ -223,13 +250,18 @@ def solve_residuals(segment):
...
 
     # process and pack
     acceleration = np.dot(D , v)
     conditions.frames.inertial.acceleration_vector = acceleration
- 
-    segment.state.residuals.acceleration_x       = np.reshape(((FT[:,0]) / m[:,0] - acceleration[:,0]),np.shape(m))
\ No newline at end of file
+    
+    if vf == 0.0: vf = 0.01
+
```
Comment (mclarke2) [inline trunk/SUAVE/Methods/Missions/Segments/Ground/Common.py:263]: and here again, table format

File: trunk/setup.py [hunk 7]
```diff
@@ -176,9 +181,14 @@ def import_tests():
         raise ImportError('scipy is required for this package')
     
     try:
-        import matplotlib.pyplot
+        import matplotlib
     except ImportError:
```
Comment (mclarke2) [inline trunk/setup.py:185]: We should also think about including Scipy. It is used even more than learn.

Comment (planes) [reply trunk/setup.py:185]: Scipy is already tested, I realized that we were missing scikit and added it

File: regression/scripts/Regional_Jet_Optimization/Procedure2.py (hunk 3 above)
Comment (planes) [reply regression/scripts/Regional_Jet_Optimization/Procedure2.py]: Done

File: trunk/SUAVE/Methods/Missions/Segments/Ground/Common.py (hunk 5 above)
Comment (planes) [reply trunk/SUAVE/Methods/Missions/Segments/Ground/Common.py:130]: Done

File: trunk/SUAVE/Methods/Missions/Segments/Ground/Common.py (hunk 6 above)
Comment (planes) [reply trunk/SUAVE/Methods/Missions/Segments/Ground/Common.py:263]: Done

File: trunk/SUAVE/Methods/Performance/estimate_landing_field_length.py (hunk 1 above)
Comment (timdmacdo) [reply trunk/SUAVE/Methods/Performance/estimate_landing_field_length.py]: Lining up SUAVE with the import and from seems odd to me, though I don't think we have any hard rules for it.

Comment (planes) [reply trunk/SUAVE/Methods/Performance/estimate_landing_field_length.py]: oh lol, I didn't put that like that in the first place. I'm obviously a hard proponent of style, but I believe it should be enforced to ensure readability. I don't think this hurts or helps.

**Pull request merged**